pkg/tsdb/prometheus/models/query.go

```diff
@@ -254,17 +254,17 @@ func interpolateVariables(
 		rateInterval = calculateRateInterval(queryInterval, requestedMinStep)
 	}
 
-	expr = strings.ReplaceAll(expr, varIntervalMs, strconv.FormatInt(int64(queryInterval/time.Millisecond), 10))
-	expr = strings.ReplaceAll(expr, varInterval, intervalv2.FormatDuration(queryInterval))
+	expr = strings.ReplaceAll(expr, varIntervalMs, strconv.FormatInt(int64(calculatedMinStep/time.Millisecond), 10))
```

Comment:
I think `calculatedMinStep` here is missleading and should be renamed to `calculatedStep`.